### ISSUE TITLE
[HOLD for payment 2023-06-29][$1000] Error message "Can't remove admin" isn't shown when selecting workspace admins

### ISSUE THREAD

**Issue by Beamanator** [Expensify/App#20809]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. go to settings -> workspaces
2.Select one workspace
3. go to members
4. try on removing admin

## Expected Result:
Error should show and you should be able to remove an admin

## Actual Result:
Admin can almost be removed, then error message is thrown

## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [ ] MacOS / Chrome / Safari
- [x] MacOS / Desktop

**Version Number:** `main`
**Reproducible in staging?:** N
**Reproducible in production?:** N
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation
**Expensify/Expensify Issue URL:**
**Issue reported by:** @ashimsharma10
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1686829788221639

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01b1012f509c1a5e12</li>
        <li>Upwork Job ID: 1670661235088379904</li>
        <li>Last Price Increase: 2023-06-19</li>
    </ul>
</details>

**Comment by Pujan92** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Error message "Can't remove admin" isn't shown when selecting workspace admins

### What is the root cause of that problem?
In the latest migration, it seems we are using `accountID` for the `WorkspaceMembersPage` component but for the `validate` method we are comparing `accountID` with the `email` and that is causing the issue.
https://github.com/Expensify/App/blob/d39643006874bf5ffec923291afa6b43884a4aec/src/pages/workspace/WorkspaceMembersPage.js#L289-L290

### What changes do you think we should make in order to solve the problem?
We need to update the condition to consider `accountID` instead of the `email` property and for error also we need to use `accountID` instead of `login`.

https://github.com/Expensify/App/blob/d39643006874bf5ffec923291afa6b43884a4aec/src/pages/workspace/WorkspaceMembersPage.js#L371-L376

**Comment by kadiealexander** [comment]
@sobitneupane could you please take a look at this issue and confirm if it's the same root cause/fix as https://github.com/Expensify/App/issues/20719 as per @dukenv0307's comment [here](https://github.com/Expensify/App/issues/20809#issuecomment-1592956218)?

**Comment by sobitneupane** [comment]
Nope @kadiealexander. It's a different issue. 

I have reviewed @Pujan92's proposal. It will solve this issue. We can go with @Pujan92's [proposal](https://github.com/Expensify/App/issues/20809#issuecomment-1592963422) to solve this issue.

**Comment by sobitneupane** [comment]
@kadiealexander I would like to take C+ role here if possible.

**Comment by kadiealexander** [comment]
Reproduced this on desktop app. @sobitneupane happy to assign you here, thanks for volunteering!

**Comment by kadiealexander** [comment]
@sobitneupane don't forget to leave the bow and eyes emoji on your review post to assign an engineer!

**Comment by kadiealexander** [comment]
@bfitzexpensify I'm going OOO til the 26th so just assigning someone to keep this moving. Happy to take it back next week!

**Comment by sobitneupane** [comment]
Thanks for the proposal.

@Pujan92's [proposal](https://github.com/Expensify/App/issues/20809#issuecomment-1592963422) looks good to me

🎀 👀 🎀 C+ reviewed

**Comment by Beamanator** [comment]
@MonilBhavsar this may technically be a deploy blocker, if we don't have one for this already 😬 So if we can move forward with this quickly (reviewing proposals & getting the PR out) that would be great!

**Comment by MonilBhavsar** [comment]
Thanks! Let's use that method and pass a single login

**Comment by MonilBhavsar** [comment]
Could you please prioritize this PR as this seems to be a deploy blocker and we want to fix it asap

**Comment by Pujan92** [comment]
> Could you please prioritize this PR as this seems to be a deploy blocker and we want to fix it asap cc @sobitneupane

Sure, will raise a PR sooner.

**Comment by MonilBhavsar** [comment]
@Beamanator let's add deploy blocker label to this issue and do you think we should CP the PR to staging?

**Comment by Beamanator** [comment]
@MonilBhavsar yes I think we would need to CP the fix to staging so that we don't break the production deploy, right? 👍

**Comment by MonilBhavsar** [comment]
Yes, correct 👍

**Comment by Pujan92** [comment]
@sobitneupane @MonilBhavsar [PR](https://github.com/Expensify/App/pull/21023) is ready for review!

**Comment by MonilBhavsar** [comment]
PR is CP'ed to staging and confirmed it is working there

**Comment by luacmartins** [comment]
Thanks for the help!

**Comment by kadiealexander** [comment]
Thanks for keeping an eye on this @bfitzexpensify!

**Comment by kadiealexander** [comment]
@sobitneupane please don't forget about the checklist!

**Comment by kadiealexander** [comment]
Paid @Pujan92 including a bonus as per [this](https://github.com/Expensify/App/issues/20809#issuecomment-1597025096)
Paid @ashimsharma10 their reporting bonus

Once the checklist is complete I'll issue C+ payment and we can close this 🎉

**Comment by sobitneupane** [comment]
> **BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:
> 
> * [ ]  [@sobitneupane] The PR that introduced the bug has been identified. Link to the PR:

https://github.com/Expensify/App/pull/20506

> * [ ]  [@sobitneupane] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:

https://github.com/Expensify/App/pull/20506#issuecomment-1617610332

> * [ ]  [@sobitneupane] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:

We were making transition (login->accountID) through multiple PRs and few cases might have missed out.

> * [ ]  [@sobitneupane] Determine if we should create a regression test for this bug.

We have decided to disable the checkbox for admin recently. It is being worked on in another issue. So, error message is no longer needed.
